packages/react-charting/src/components/HorizontalBarChart/HorizontalBarChart.base.tsx

```diff
@@ -288,19 +295,47 @@ export class HorizontalBarChartBase extends React.Component<IHorizontalBarChartP
     );
   }
 
+  // Adjusts the margin percentage, based on the width of the svg (inversly propotional)
+  private _adjustBarSpacing(): void {
+    const svgWidth = this.barChartSvgRef.current?.getBoundingClientRect().width;
+    if (svgWidth) {
+      // [0,1] Controls the spacing between the bars
+      const BAR_SPACING_FACTOR = 1;
+      // The barspacing is provided in percentage terms, so changes inversely to the width of the svg
+      const currentBarSpacing = clamp((200 / svgWidth) * BAR_SPACING_FACTOR, 1, 0.2);
+      this.setState({ barSpacing: currentBarSpacing });
+    }
+  }
+
+  /**
+   * This functions returns an array of <rect> elements, which form the bars
+   * For each bar an x value, and a width needs to be specified
+   * The computations are done based on percentages
+   * Extra margin is also provided, in the x value to provide some spacing in between the bars
```

Comment:
Add 10 bars with value x=0.5 and check if it renders properly.

Reply:
![image](https://github.com/microsoft/fluentui/assets/70366079/b46818eb-9809-4d18-9b41-49283e538ce9)

![image](https://github.com/microsoft/fluentui/assets/70366079/dfe1f7a7-a54d-420f-bef5-0da1657e7ef5)

![image](https://github.com/microsoft/fluentui/assets/70366079/a89e6840-11a1-4c26-941d-9a7f51b8a58c)

![image](https://github.com/microsoft/fluentui/assets/70366079/e7a4cb4c-cb1a-418b-bbe4-ec6c09f49d7b)

Reply:
based on the absolute or relative scale

Reply:
For the relative scale add these 0.5 at the end of the existing bar and verify if it works properly.

Reply:
![image](https://github.com/microsoft/fluentui/assets/70366079/995e85a7-7c08-4379-b29c-430087925d70)

It seems like its exceeding the width a little

Reply:
![image](https://github.com/microsoft/fluentui/assets/70366079/9d9c4fc2-04a4-4c83-a006-a12e5d393051)

![image](https://github.com/microsoft/fluentui/assets/70366079/0978129c-9680-420f-b071-6eef4678832e)

The width overshoots, need to fix this one

Reply:
![image](https://github.com/microsoft/fluentui/assets/70366079/cd3848ef-db82-4b4b-9a72-bb327cfa892d)

Fixed this one, by adjusting the scaling ratio correctly